fili-core/src/test/groovy/com/yahoo/bard/webservice/web/DataApiRequestSortSpec.groovy

```diff
@@ -0,0 +1,142 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web
+
+import com.yahoo.bard.webservice.application.JerseyTestBinder
+import com.yahoo.bard.webservice.druid.model.orderby.OrderByColumn
+import com.yahoo.bard.webservice.druid.model.orderby.SortDirection
+import com.yahoo.bard.webservice.web.endpoints.DataServlet
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class DataApiRequestSortSpec extends Specification {
+
+    JerseyTestBinder jtb
+
+    def setup() {
+        // Create the test web container to test the resources
+        jtb = new JerseyTestBinder(DataServlet.class)
+    }
+
+    def cleanup() {
+        // Release the test web container
+        jtb.tearDown()
+    }
+
+
+
+    def "If dateTime is not the first value in the sort list, then throw an error"() {
+        setup:
+        String expectedMessage = ErrorMessageFormat.DATE_TIME_SORT_VALUE_INVALID.format()
+
+        when:
+        new DataApiRequest().generateDateTimeSortColumn(["xyz":SortDirection.DESC,"dateTime":SortDirection.DESC])
+
+        then:
+        Exception e = thrown(BadApiRequestException)
+        e.getMessage() == expectedMessage
+    }
+
+    @Unroll
+    def "Validate the sort column and direction map from #sortString string"() {
+        when:
+        Map<String, SortDirection> actual = new DataApiRequest().generateSortColumns(sortString)
+
+        then:
+        actual == expected
```

Comment:
Can be simplified to just an `expected` block, since we're just comparing a value to a single call.